pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+  * [About Pipeline YAML](#pipeline-yaml-reference)
+  * [The preamble](#the-preamble)
+    * [version](#version)
+    * [name](#name)
+    * [agent](#agent)
+    * [Preamble example](#preamble-example)
+  * [Blocks](#blocks)
+  * [Task](#task)
+  * [Example of task](#example-of-task)
+  * [Jobs](#jobs)
+  * [Example of jobs](#example-of-jobs)
+  * [The secrets property](#the-secrets-property)
+  * [Example of secrets](#example-of-secrets)
+  * [Complete examples](#complete-configuration-examples)
+  * [The order of execution](#the-order-of-execution)
+  * [See also](#see-also)
+
+
+# Pipeline YAML Reference
+
+This document is the reference for the YAML grammar used for describing the
+pipelines of Semaphore 2.0.
+
+The core properties of a Semaphore 2.0 pipeline configuration file are `blocks`,
+which appear only once at the beginning of a YAML file, `task`, which can
+appear multiple times and `jobs`, which can also be repeated.
+
+
+#### Comments
+
+Lines that begin with `#` are considered comments and are being ignored by the
+YALM parser, which is not a Semaphore 2.0 feature but the way YAML files work.
+
+
+## The preamble
+Each Semaphore 2.0 pipeline configuration file has a preamble, which is
+mandatory and should be present on every `.semaphore/semaphore.yml` file you create.
+
+### Properties
+
+### version
+
+The `version` property is a string value that shows the version of the
+pipeline YAML file that will be used.
+
+List of valid values: `"v1.0"`
+
+Example: `version: "v1.0"`
+
+### name
+
+The `name` property is a Unicode string that assigns a name to the Semaphore
+2.0 project and is optional.
+
+*Caution*: The `name` property can be found in other sections because it
+is used for defining the name of a job inside a `jobs` block or the name of
+a `task` section defined using `task`.
+
+Example: `name: "The name of the Semaphore 2.0 project"`
+
+
+### agent
+
+The `agent` property requires three additional properties that identify the
+environment, that is the hardware of the Virtual machine and the Operating
+System of the Virtual Machine, in which the jobs of the pipeline will get
+executed.
+
+Example:
+
+    agent:
+      machine:
+        type: e1-standard-2
+        os_image: ubuntu1804
+
+
+#### machine
+
+The `machine` property, which can only be found under `agent`, holds the `type`
+and `os_image` properties.
+
+Example:
+
+    machine:
+      type: e1-standard-2
+      os_image: ubuntu1804
+
+
+#### type
+
+The `type` property is intended for selecting the machine (hardware) you would
+like to use for your builds.
+
+List of valid values: `e1-standard-2`
+
+Example: `type = e1-standard-2`
+
+
+#### os_image
+
+The `os_image` property specifies the operating system and the version of the
+operating system that will be used.
+
+
+Valid values: `ubuntu1804`
+
+
+Example: `os_image = ubuntu1804`
+
+#### Constants
```

Comment:
Move these constants under `type` and `os_image`. There is no need to be overly formal and keep the outside of the context where they are used.

Reply:
In that case, I will also remove the `#### Constants` line because otherwise it will appear multiple times (one under `type` and another under `os_image`).